### PR TITLE
Visualize `RPNGDescription` and `RPNGTemplate`

### LINKED ISSUE #425: Visualize `RPNGTemplate` instances
**Is your feature request related to a problem? Please describe.**

Visualization of the `RPNGTemplate` instances will be helpful for both the developers and users to inspect and understand the underlying circuit structures.

**Describe the solution you'd like**

An SVG-based visualization would be great. The design can resemble those found in Austin's review paper and slides, with minor modifications to include r/g information. Specifically:

- Use colors at the corners to indicate interaction bases.
- Display numbers to represent the ordering.
- Maybe include small circles with annotated bases at the corners to indicate the r/g fields.
- Use lines in the plaquettes to indicate hook errors.

### PR DESCRIPTION
**Pull request by inmzhang**
Closes #425

This PR implements `rpng_svg_viewer` function which takes in a single `RNGDescription` or a 2D list of `RNGDescription` and returns the SVG string. An example of the visualization of a spatial junction template is as below:

图片

`R`/`G` fields are represented by rect/circles on the data qubits: in the above exmaple, the blue rects on the data qubits represent Z basis resets, the red circles represent X basis measurements.

### CODE REVIEW

File: main.py
Comment (nelimee) [inline main.py:1]: Probably not meant to be included in the PR.

File: src/tqec/templates/rpng.py [hunk 1]
```diff
@@ -79,3 +81,53 @@ def instantiation_origin(self, k: int) -> Position2D:
...
+                corresponds to the basis. The G field is shown as a small circle at the position
+                of the data qubit, whose color corresponds to the basis.
+            show_plaquette_indices: Whether to show the indices of the plaquettes. If True, the
+                indices are shown at the center of the plaquettes.
+            show_interaction_order: Whether to show the interaction order of the plaquettes. If
+                True, the interaction order is shown at each corner of the plaquette.
+            show_hook_error: A predicate function that takes an RPNGDescription and returns True
+                if the plaquette should be highlighted with the hook error. The hook error is
+                shown as a black line along the hook edge.
+
```
Comment (nelimee) [inline src/tqec/templates/rpng.py:113]:
```suggestion
```

**Review by nelimee — COMMENTED** [review]
Well done, the visualisation is super nice!

The code performing the drawing is a little bit hard to follow, but the visual result seems correct. One minor nitpick: the text does not seem to be correctly aligned vertically:

- plaquette schedule for top qubits seems too high when compared to the bottom ones,
- the plaquette index also seem too high and not vertically centered.

*Maybe* [setting `dominant-baseline` to `central` instead of `middle`](https://stackoverflow.com/questions/5546346/how-to-place-and-center-text-in-an-svg-rectangle)?

**Comment by inmzhang** [comment]
Thanks for the quick review. Merry Christmas!

**Comment by nelimee** [comment]
> Thanks for the quick review. Merry Christmas!

Merry Christmas to you too :-) Is using `central` fixing the alignment issue?

**Comment by inmzhang** [comment]
> Is using `central` fixing the alignment issue?

I think so. The following code generates the image below:

```py
from tqec.compile.specs.enums import JunctionArms
from tqec.templates.library import get_spatial_junction_qubit_template

template = get_spatial_junction_qubit_template('x', JunctionArms.UP | JunctionArms.RIGHT)

template.view_as_svg(k=3, write_to_filepath="test.svg")
```

**Review by nelimee — APPROVED** [review]
LGTM, well done, this is super useful to visualise what we implement!
